SET OPTION is removed in mysql 5.6:

- https://bugs.mysql.com/bug.php?id=66765
- https://bugs.mysql.com/bug.php?id=67507
- https://www.markotomic.com/mysqldump-mysql-5-6-problem-solved/

The change works with mysql 5.0 server as well:

Server version: 5.0.96-log MySQL Community Server (GPL)

mysql> SET SQL_QUOTE_SHOW_CREATE=1;
Query OK, 0 rows affected (0.00 sec)

mysql> SET OPTION SQL_QUOTE_SHOW_CREATE=1;
Query OK, 0 rows affected (0.00 sec)

--- mysql-5.0.96/client/mysqldump.c~	2020-11-27 15:09:05.000000000 +0200
+++ mysql-5.0.96/client/mysqldump.c	2020-11-27 15:14:43.247336010 +0200
@@ -1708,7 +1708,7 @@
   verbose_msg("-- Retrieving table structure for table %s...\n", table);
 
   len= my_snprintf(query_buff, sizeof(query_buff),
-                   "SET OPTION SQL_QUOTE_SHOW_CREATE=%d",
+                   "SET SQL_QUOTE_SHOW_CREATE=%d",
                    (opt_quoted || opt_keywords));
   if (!create_options)
     strmov(query_buff+len,
@@ -3753,7 +3753,7 @@
   verbose_msg("-- Retrieving view structure for table %s...\n", table);
 
 #ifdef NOT_REALLY_USED_YET
-  sprintf(insert_pat,"SET OPTION SQL_QUOTE_SHOW_CREATE=%d",
+  sprintf(insert_pat,"SET SQL_QUOTE_SHOW_CREATE=%d",
           (opt_quoted || opt_keywords));
 #endif
 
